### ISSUE TITLE
Errors Compiling

### ISSUE THREAD

**Issue by ParkerFleming** [pblesi/touch_light#5]
So I'm a bit of a novice but got myself to the point of compiling the code and get this set of errors.

I'm running the "particle compile photon" command from the src directory. That directory has just the touch_light.ino and wifi_creds.h files in it (I commented out the wifi_creds.h from the code since I will need the recipient to configure the wifi on their own via the mobile app).

Obviously neopixel.h isn't in the directory, so I'm wondering if the instructions missed a step to pull that down and put it in the dir?

touch_light.cpp:10:22: fatal error: neopixel.h: No such file or directory
compilation terminated.
../build/module.mk:267: recipe for target '../build/target/user/platform-6-mtouch_light.o' failed
make[2]: Leaving directory '/firmware/user'
make[2]: *** [../build/target/user/platform-6-mtouch_light.o] Error 1
../../../build/recurse.mk:11: recipe for target 'user' failed
make[1]: *** [user] Error 2
make[1]: Leaving directory '/firmware/modules/photon/user-part'
../build/recurse.mk:11: recipe for target 'modules/photon/user-part' failed
make: *** [modules/photon/user-part] Error 2
Compile failed: Compiler encountered an error

Any thoughts on what could be going wrong?

**Comment by pblesi** [comment]
@ParkerFleming 

I think the problem has to do with where you are running the compile command from. You should not be in the `src` directory, but one directory up from the src directory.

Let me know if that helps!

**Comment by ParkerFleming** [comment]
Thanks! That got my code to compile. My new issues is that I've flashed the code to both of my Photons, and one will connect and stay connected to WiFi and particle's cloud, the other cycles from Cyan (Connected to WiFi + Particle Cloud) to pulsing Green (Connected to WiFi but not connected to the Particle Cloud).

I've read through: https://docs.particle.io/tutorials/device-os/led/photon/#cloud-not-connected but I'm out of my depth debugging this and also don't understand why one of these Photons would work perfectly and the other not with the same compiled code on both of them.

Any thoughts?

(BTW, My family and I are super appreciative of this project and your help!!!)

**Comment by pblesi** [comment]
@ParkerFleming ,

Does the light flash white and then go dark, never getting to the rainbow color wheel phase? If so, it is likely an issue reading the D3 pin. The way that the touch sensor works is that it sets the D4 pin to high and then measures how long it takes for the D3 pin to read high. If D3 never reads high, then the user code can get in an infinite loop waiting for it to go high (preventing the cloud-connect code from running).

One thing you can do to test this is flashing the tinker app onto the device and then test setting the D4 pin to high and attempting to read on the D3 pin. You can also try this by shorting the resistor and directly connecting these two pins to see if you get a different result.

If this is the case there are a few things you can do. You can recheck the solder joints for the resistor to ensure they are good and solid. You could try a different resistor to see if it works better. I have run into this issue in the past and have switched to using D5 as the read pin instead of D3. For whatever reason, this has seemed to work better with some photons.

If you do switch to use the D5 pin, you will change the code at [this location](https://github.com/pblesi/touch_light/blob/master/src/touch_light.ino#L101) to be "D5":

Hope this helps!

**Comment by chenseachels** [comment]
I've followed all the steps up until connecting all the wiring. For some reason, whenever I connect both VIN and GND to PWR +5V and GND respectively, the light on the photon goes out. I'm assuming somehow it not enough power? but i'm not sure if theres something else going on. I also tried getting the .bin file flashed onto the photon but as soon as I flash, the photon starts blinking the SOS blinks. Any help would be appreciated
